primitives/runtime/src/traits.rs

```diff
@@ -1216,9 +1216,36 @@ macro_rules! impl_opaque_keys {
 				pub $field:ident: $type:ty,
 			)*
 		}
+	) => {
+		$crate::paste::paste!(
+			$crate::impl_opaque_keys!(@with_serde_crate_literal
+				$( #[ $attr ] )*
+				pub struct $name {
+					$(
+						$( #[ $inner_attr ] )*
+						pub $field: $type,
+					)*
+				}
+				// NOTE: we abuse from `paste` feature which concatenate string in docs to build
+				// our string
+				#[doc = "__opaque_keys_serde_import__" $name]
```

Comment:
Ahh but wait. We don't need to do this soo complicated

Reply:
ah great